core/storage/plugin_unix.go

```diff
@@ -8,20 +8,46 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"path/filepath"
 	"strings"
+	"sync"
 
 	"github.com/docker/go-connections/sockets"
 	"github.com/leodotcloud/log"
 	"github.com/pkg/errors"
 	"github.com/rancher/agent/model"
 )
 
-func CallRancherStorageVolumePlugin(volume model.Volume, action string, payload interface{}) (Response, error) {
+var client = &http.Client{}
+var transportMap = &transportStore{
+	tMap: map[string]*http.Transport{},
+	lock: sync.RWMutex{},
+}
+
+type transportStore struct {
+	tMap map[string]*http.Transport
+	lock sync.RWMutex
+}
+
+func (t *transportStore) add(driver string) {
+	t.lock.Lock()
+	defer t.lock.Unlock()
 	transport := new(http.Transport)
```

Comment:
you could hit a race condition where there are two back to back calls to add() and the second one gets in and overrides what the first one set.
To fix, add another check for the transport inside add, after you obtain the write lock.

Suggestion (take it or leave it): could you also have add return the transport, thus eliminatuing the second call to get?